### PR TITLE
Refactor the Helm chart

### PR DESCRIPTION
**Pull request by jakalkhalili**

### CODE REVIEW

File: chart/jenkins-operator/templates/jenkins-cr.yaml [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
   name: {{ .Values.jenkins.name }}
+  namespace: {{ .Values.jenkins.namespace }}
 spec:
-  {{- toYaml .Values.jenkins.spec | nindent 4 }}
+  master:
+    {{- with .Values.jenkins.plugins }}
+    plugins: {{ toYaml . | nindent 4 }}
+    {{- end }}
+    {{- with .Values.jenkins.notifications }}
+    notifications: {{ toYaml . | nindent 4 }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins-cr.yaml]: it is not under master is under spec

File: chart/jenkins-operator/templates/jenkins-cr.yaml [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+    {{- end }}
+    containers:
+      - name: jenkins-master
+        image: jenkins/jenkins:lts
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins-cr.yaml]: must be configurable

File: chart/jenkins-operator/values.yaml [hunk 3]
```diff
@@ -15,56 +15,29 @@ fullnameOverride: ""
...
-            requests:
-              cpu: "1"
-              memory: 500Mi
+  metadata:
+  name: jenkins
+  namespace: default
+  imagePullSecrets: []
+  notifications: []
+  plugins:
+    - name: simple-theme-plugin
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]: don't install it by default, show usage in comments

File: chart/jenkins-operator/values.yaml [hunk 4]
```diff
@@ -15,56 +15,29 @@ fullnameOverride: ""
...
+      version: 0.5.1
+  seedJobs:
+  - id: jenkins-operator
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]: comment it

File: chart/jenkins-operator/templates/backup.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{{ if .Values.backup.enabled }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.jenkins.name }}-backuppvc
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/backup.yaml]:
```suggestion
  name: {{ .Values.jenkins.name }}-backup
```

File: chart/jenkins-operator/templates/backup.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  namespace: {{ .Values.jenkins.namespace }}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: {{ .Values.backup.storageSize }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/backup.yaml]: allow set storage class

File: chart/jenkins-operator/templates/backup.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{{ if .Values.backup.enabled }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/backup.yaml]: not all backups will use PVC

File: chart/jenkins-operator/Chart.yaml [hunk 8]
```diff
@@ -2,5 +2,5 @@ apiVersion: v1
 appVersion: "0.3.3"
 description: Kubernetes native operator which fully manages Jenkins on Kubernetes
 name: jenkins-operator
-version: 0.0.8
+version: 0.0.9
```
Comment (tomaszsek) [inline chart/jenkins-operator/Chart.yaml]:
```suggestion
version: 0.1.0
```

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 9]
```diff
@@ -0,0 +1,92 @@
+{{ if .Values.jenkins.enabled }}
+apiVersion: {{ .Values.jenkins.apiVersion }}
+kind: Jenkins
+metadata:
+  name: {{ .Values.jenkins.name }}
+  namespace: {{ .Values.jenkins.namespace }}
+spec:
+  disableCSRFProtection: {{ .Values.jenkins.disableCSRFProtection }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: wrong place, move under `master`

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    {{- if .Values.backup.recoveryOnce }}
+    recoveryOnce: {{ .Values.backup.recoveryOnce }}
+    {{- end }}
+  {{- end }}
+  {{- with .Values.jenkins.plugins }}
+  plugins: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  master:
+    {{- with .Values.jenkins.notifications }}
+    notifications: {{ toYaml . | nindent 4 }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: wrong place, move under `spec`

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+          successThreshold: 1
+          timeoutSeconds: 1
+        {{- with .Values.jenkins.resources }}
+        resources: {{ toYaml . | nindent 10 }}
+        {{- end }}
+      {{- if .Values.backup.enabled }}
+      - name: {{ .Values.backup.containerName }}
+        env:
+          - name: BACKUP_DIR
+            value: {{ .Values.backup.backupDir }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: env related only to the specific backup provider

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+          - name: JENKINS_HOME
+            value: {{ .Values.backup.jenkinsHome }}
+          - name: BACKUP_COUNT
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: env related only to the specific backup provider

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+            value: {{ .Values.backup.backupCount | quote }}
+        image: {{ .Values.backup.image }}
+        imagePullPolicy: IfNotPresent
+        volumeMounts:
+          - mountPath: {{ .Values.backup.jenkinsHome }}
+            name: jenkins-home
+          - mountPath: {{ .Values.backup.backupDir }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: env related only to the specific backup provider

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+            name: backup
+    volumes:
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: volumes are a global thing not only related to backup

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+      - name: backup
+        persistentVolumeClaim:
+          claimName: {{ .Values.jenkins.name }}-backuppvc
+    {{- end }}
+
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: missing configuration sections: groovy scripts & CasC

File: chart/jenkins-operator/values.yaml [hunk 16]
```diff
@@ -2,69 +2,67 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
-replicaCount: 1
+backup:
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]: should be under `jenkins` section

File: chart/jenkins-operator/templates/backup.yaml [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
```
Comment (rdemachkovych) [inline chart/jenkins-operator/templates/backup.yaml]: it will be a good point add `labels` to each k8s object

Comment (rdemachkovych) [reply chart/jenkins-operator/templates/backup.yaml]: this makes sense if the operator does not do this

File: chart/jenkins-operator/templates/jenkins.yaml (hunk 11 above)
Comment (jakalkhalili) [reply chart/jenkins-operator/templates/jenkins.yaml]: what do you mean? what is "specific backup provider"

File: chart/jenkins-operator/values.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (jakalkhalili) [reply chart/jenkins-operator/values.yaml]: moved under `jenkins` section

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+apiVersion: {{ .Values.jenkins.apiVersion }}
+kind: Jenkins
+metadata:
+  name: {{ .Values.jenkins.name }}
+  namespace: {{ .Values.jenkins.namespace }}
+spec:
+  {{- with .Values.jenkins.configuration.configurationAsCode }}
+  configurationAsCode: {{ toYaml . | nindent 8 }}
+  {{- end }}
+  {{- with .Values.jenkins.configuration.groovyScriots }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]:
```suggestion
  {{- with .Values.jenkins.configuration.groovyScripts }}
```

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+        {{- end }}
+    {{- if .Values.jenkins.backup.recoveryOnce }}
+    recoveryOnce: {{ .Values.jenkins.backup.recoveryOnce }}
+    {{- end }}
+  {{- end }}
+  {{- with .Values.jenkins.notifications }}
+  notifications: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.jenkins.plugins }}
+  plugins: {{ toYaml . | nindent 4 }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: plugins should be under `spec.master`

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+  {{- end }}
+  master:
+    disableCSRFProtection: {{ .Values.jenkins.disableCSRFProtection }}
+    containers:
+      - name: jenkins-master
+        image: {{ .Values.jenkins.image }}
+        imagePullPolicy: Always
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: parametrize

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+      - name: {{ .Values.jenkins.backup.containerName }}
+        {{- with .Values.jenkins.backup.env }}
+        env: {{- toYaml . | nindent 4 }}
+        {{- end }}
+        image: {{ .Values.jenkins.backup.image }}
+        imagePullPolicy: IfNotPresent
+        {{- with .Values.jenkins.backup.volumeMounts }}
+        volumeMounts: {{- toYaml . | nindent 4 }}
+        {{- end }}
+    {{- with .Values.jenkins.backup.volumes }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]:
```suggestion
    {{- with .Values.jenkins.volumes }}
```

File: chart/jenkins-operator/values.yaml [hunk 22]
```diff
@@ -2,69 +2,88 @@
...
+  resources:
+    limits:
+      cpu: 1500m
+      memory: 3Gi
+    requests:
+      cpu: 1
+      memory: 500Mi
+
+  backup:
+    enabled: false
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]:
```suggestion
    enabled: true
```

File: chart/jenkins-operator/values.yaml [hunk 23]
```diff
@@ -2,69 +2,88 @@
...
+    image: virtuslab/jenkins-operator-backup-pvc:v0.0.7
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]:
```suggestion
    image: virtuslab/jenkins-operator-backup-pvc:v0.0.8
```

File: chart/jenkins-operator/values.yaml [hunk 24]
```diff
@@ -2,69 +2,88 @@
...
+      size: 5Gi
+      className: ""
+
+    env:
+      - name: BACKUP_DIR
+        value: /backup
+      - name: JENKINS_HOME
+        value: /jenkins-home
+      - name: BACKUP_COUNT
+        value: "3" # keep only the 2 most recent backups
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]:
```suggestion
        value: "3" # keep only the 3 most recent backups
```

File: chart/jenkins-operator/values.yaml [hunk 25]
```diff
@@ -2,69 +2,88 @@
...
+        persistentVolumeClaim:
+          claimName: jenkins-backup
+    volumeMounts:
+      - name: jenkins-home
+        mountPath: /jenkins-home # Jenkins home volume
+      - mountPath: /backup # backup volume
+        name: backup
+
+  configuration:
+    configurationAsCode: {}
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]: you have to create config maps as well

File: chart/jenkins-operator/values.yaml [hunk 26]
```diff
@@ -2,69 +2,88 @@
...
+    groovyScript: {}
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]: you have to create config maps as well

File: chart/jenkins-operator/values.yaml [hunk 27]
```diff
@@ -2,69 +2,88 @@
...
+      - name: backup # PVC volume where backups will be stored
+        persistentVolumeClaim:
+          claimName: jenkins-backup
+    volumeMounts:
+      - name: jenkins-home
+        mountPath: /jenkins-home # Jenkins home volume
+      - mountPath: /backup # backup volume
+        name: backup
+
+  configuration:
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]: don't forget about secrets for groovy scripts and CasC

File: chart/jenkins-operator/templates/configuration.yaml [hunk 28]
```diff
@@ -0,0 +1,20 @@
+{{ if .Values.jenkins.configuration.configurationAsCode.content }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: jenkins-operator-casc-user-configuration-{{ .Values.jenkins.name }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/configuration.yaml]:
```suggestion
  name: jenkins-{{ .Values.jenkins.name }}-casc
```

File: chart/jenkins-operator/templates/configuration.yaml [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+  namespace: {{ .Values.jenkins.namespace }}
+{{- with .Values.jenkins.configuration.configurationAsCode.content }}
+data: {{ toYaml . | nindent 2 }}
+{{- end }}
+{{ end }}
+{{ if .Values.jenkins.configuration.groovyScript.content }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/configuration.yaml]:
```suggestion
{{ if .Values.jenkins.configuration.groovyScripts.content }}
```

File: chart/jenkins-operator/templates/configuration.yaml [hunk 30]
```diff
@@ -0,0 +1,20 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: jenkins-operator-gs-user-configuration-{{ .Values.jenkins.name }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/configuration.yaml]:
```suggestion
  name: jenkins-{{ .Values.jenkins.name }}-groovy-scripts
```

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 31]
```diff
@@ -0,0 +1,94 @@
...
+apiVersion: {{ .Values.jenkins.apiVersion }}
+kind: Jenkins
+metadata:
+  name: {{ .Values.jenkins.name }}
+  namespace: {{ .Values.jenkins.namespace }}
+spec:
+  {{- if .Values.jenkins.configuration.configurationAsCode.content }}
+  configurationAsCode:
+    configurations:
+    - name: jenkins-operator-casc-user-configuration-{{ .Values.jenkins.name }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: missing https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/schema/#github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2.SecretRef

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+  {{- end }}
+  {{- if .Values.jenkins.configuration.groovyScript.content }}
+  groovyScripts:
+    configurations:
+    - name: jenkins-operator-gs-user-configuration-{{ .Values.jenkins.name }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: missing missing https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/schema/#github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2.SecretRef

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 33]
```diff
@@ -0,0 +1,94 @@
...
+        {{- end }}
+    {{- if .Values.jenkins.backup.recoveryOnce }}
+    recoveryOnce: {{ .Values.jenkins.backup.recoveryOnce }}
+    {{- end }}
+  {{- end }}
+  {{- with .Values.jenkins.notifications }}
+  notifications: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  master:
+    {{- with .Values.jenkins.plugins }}
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]: missing base plugins

File: chart/jenkins-operator/values.yaml (hunk 27 above)
Comment (tomaszsek) [reply chart/jenkins-operator/values.yaml]: you have to give ability to add secrets values here

Comment (tomaszsek) [reply chart/jenkins-operator/values.yaml]: secrets value not secret!

Comment (tomaszsek) [reply chart/jenkins-operator/values.yaml]: and also you have to create that secret

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 34]
```diff
@@ -0,0 +1,109 @@
...
+  {{- end }}
+  {{- if .Values.jenkins.configuration.groovyScripts.content }}
+  groovyScripts:
+    configurations:
+    - name: jenkins-{{ .Values.jenkins.name }}-gs
+    secret:
+      {{- if .Values.jenkins.configuration.groovyScripts.secretRefName }}
+      name: {{ .Values.jenkins.configuration.groovyScripts.secretRefName }}
+      {{ else if .Values.jenkins.configuration.groovyScripts.secretData }}
+      name: jenkins-{{ .Values.jenkins.name }}-gs
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]:
```suggestion
      name: jenkins-{{ .Values.jenkins.name }}
```

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 35]
```diff
@@ -0,0 +1,109 @@
...
+spec:
+  {{- if .Values.jenkins.configuration.configurationAsCode.content }}
+  configurationAsCode:
+    configurations:
+    - name: jenkins-{{ .Values.jenkins.name }}-casc
+        secret:
+      {{- if .Values.jenkins.configuration.configurationAsCode.secretRefName }}
+      name: {{ .Values.jenkins.configuration.configurationAsCode.secretRefName }}
+      {{ else if .Values.jenkins.configuration.configurationAsCode.secretData }}
+      name: jenkins-{{ .Values.jenkins.name }}-casc
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/jenkins.yaml]:
```suggestion
      name: jenkins-{{ .Values.jenkins.name }}
```

File: chart/jenkins-operator/values.yaml [hunk 36]
```diff
@@ -2,69 +2,99 @@
...
+  imagePullPolicy: IfNotPresent
+
+  imagePullSecrets: []
+  nameOverride: ""
+  fullnameOverride: ""
+
+  resources: {}
+  nodeSelector: {}
+  tolerations: []
+  affinity: {}
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml]: move at the beginning of the file

File: chart/jenkins-operator/values.yaml [hunk 37]
```diff
@@ -1,32 +1,144 @@
-# Default values for jenkins-operator.
-# This is a YAML-formatted file.
-# Declare variables to be passed into your templates.
-
+# Jenkins Custom Resource (CR) Helm chart
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:1]: Jenkins Operator Helm chart

File: chart/jenkins-operator/values.yaml [hunk 38]
```diff
@@ -1,32 +1,144 @@
...
 
+# jenkins configures pod instance
 jenkins:
+  # enabled can enable or disable the jenkins instance
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:5]: jenkins -> Jenkins

File: chart/jenkins-operator/values.yaml [hunk 39]
```diff
@@ -1,32 +1,144 @@
...
+  # Set to false if you have configured CR already and/or you want to deploy only operator
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:6]: operator only

File: chart/jenkins-operator/values.yaml [hunk 40]
```diff
@@ -68,28 +209,44 @@ jenkins:
...
     secretRefName: ""
+
+    # secretData creates new secret if secretRefName is empty and fills with data provided in secretData
     secretData: {}
 
+# operator is section for configuring operator deployment
 operator:
   replicaCount: 1
 
+  # image used as operator
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:237]: image is the name (and tag) of the Jenkins Operator image

File: chart/jenkins-operator/values.yaml [hunk 41]
```diff
@@ -68,28 +209,44 @@ jenkins:
...
+  # imagePullPolicy defines policy for pulling images
   imagePullPolicy: IfNotPresent
 
+  # imagePullSecrets is used if you want to pull images from private repository
   imagePullSecrets: []
+
+  # nameOverride overrides the deployment name
   nameOverride: ""
+
+  # nameOverride overrides the deployment name
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:249]: fullnameOverride overrides the ?

File: chart/jenkins-operator/values.yaml [hunk 42]
```diff
@@ -1,32 +1,144 @@
-# Default values for jenkins-operator.
-# This is a YAML-formatted file.
-# Declare variables to be passed into your templates.
-
+# Jenkins Custom Resource (CR) Helm chart
 
+# jenkins configures pod instance
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:3]: jenkins configures pod instance - > Jenkins instance configuration

File: chart/jenkins-operator/values.yaml [hunk 43]
```diff
@@ -1,32 +1,144 @@
...
 jenkins:
+  # enabled can enable or disable the jenkins instance
+  # Set to false if you have configured CR already and/or you want to deploy only operator
   enabled: true
 
+  # apiVersion is version of the CR manifest template
+  # The recommended and default value is "jenkins.io/v1alpha2"
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:10]: please add a link to the documentation with available api versions

File: chart/jenkins-operator/values.yaml [hunk 44]
```diff
@@ -1,32 +1,144 @@
...
+  # name of resource
+  # The pod name will be jenkins-<name> (name will be set as suffix)
   name: jenkins
+
+  # namespace is the namespace where the resources will be deployed
+  # It's not recommended to use default namespace
+  # Create new namespace for jenkins (called e.g. jenkins)
   namespace: default
 
+  # image used by resource
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:22]: image is the name (and tag) of the Jenkins instance

File: chart/jenkins-operator/values.yaml [hunk 45]
```diff
@@ -1,32 +1,144 @@
...
+  #     - loggingLevel: info
+  #       verbose: true
+  #       name: <name>
+  #       slack:
+  #         webHookURLSecretKeySelector:
+  #           secret:
+  #             name: <secret_name>
+  #           key: <key>
+  #
+  # # Teams
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:58]: Microsoft Teams

File: chart/jenkins-operator/values.yaml [hunk 46]
```diff
@@ -1,32 +1,144 @@
...
+  #         apiKeySecretKeySelector:
+  #           secret:
+  #             name: <secret_name>
+  #           key: <key>
+  #         recipient: <your_email>
+  #         from: <mailgun_email>
+  #
   notifications: []
 
+  # basePlugins are plugins installed by operator
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:91]: please add a note this section shouldn't be used for custom plugins

File: chart/jenkins-operator/values.yaml [hunk 47]
```diff
@@ -1,32 +1,144 @@
...
+  #    repositoryBranch: master
+  #    repositoryUrl: https://github.com/jenkinsci/kubernetes-operator.git
   seedJobs: []
-#  - id: jenkins-operator
-#    targets: "cicd/jobs/*.jenkins"
-#    description: "Jenkins Operator repository"
-#    repositoryBranch: master
-#    repositoryUrl: https://github.com/jenkinsci/kubernetes-operator.git
+
+  # resources used by Jenkins
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:140]: resources - > Resource limits for Jenkins instance

File: chart/jenkins-operator/values.yaml [hunk 48]
```diff
@@ -68,28 +209,44 @@ jenkins:
...
+  # imagePullSecrets is used if you want to pull images from private repository
   imagePullSecrets: []
+
+  # nameOverride overrides the deployment name
   nameOverride: ""
+
+  # nameOverride overrides the deployment name
   fullnameOverride: ""
 
   resources: {}
```
Comment (antoniaklja) [inline chart/jenkins-operator/values.yaml:252]: what's that?

File: chart/jenkins-operator/values.yaml [hunk 49]
```diff
@@ -1,70 +1,255 @@
...
-fullnameOverride: ""
+# Jenkins Custom Resource (CR) Helm chart
 
+# jenkins configures pod instance
 jenkins:
+  # enabled can enable or disable the jenkins instance
+  # Set to false if you have configured CR already and/or you want to deploy only operator
   enabled: true
+
+  # apiVersion is version of the CR manifest template
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:9]:
```suggestion
  # apiVersion is the version of the CR manifest
```

File: chart/jenkins-operator/values.yaml (hunk 43 above)
Comment (tomaszsek) [reply chart/jenkins-operator/values.yaml:10]:
```suggestion
  # link to the scheme
```

File: chart/jenkins-operator/values.yaml [hunk 50]
```diff
@@ -1,70 +1,255 @@
...
+
+  # notifications is feature that notify user about Jenkins reconcilation status
+  # See https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/notifications/ for more info
+  #
+  # Example:
+  #
+  # # Slack
+  # kind: Jenkins
+  # spec:
+  #   master:
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:48]: remove

File: chart/jenkins-operator/values.yaml [hunk 51]
```diff
@@ -1,70 +1,255 @@
...
+  #       slack:
+  #         webHookURLSecretKeySelector:
+  #           secret:
+  #             name: <secret_name>
+  #           key: <key>
+  #
+  # # Teams
+  # kind: Jenkins
+  # spec:
+  #   master:
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:62]: remove

File: chart/jenkins-operator/values.yaml [hunk 52]
```diff
@@ -1,70 +1,255 @@
...
+  #       teams:
+  #         webHookURLSecretKeySelector:
+  #           secret:
+  #             name: <secret_name>
+  #           key: <key>
+  #
+  # # Mailgun
+  # kind: Jenkins
+  # spec:
+  #   master:
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:76]: remove

File: chart/jenkins-operator/values.yaml [hunk 53]
```diff
@@ -1,70 +1,255 @@
...
+  #         apiKeySecretKeySelector:
+  #           secret:
+  #             name: <secret_name>
+  #           key: <key>
+  #         recipient: <your_email>
+  #         from: <mailgun_email>
+  #
+  notifications: []
+
+  # basePlugins are plugins installed by operator
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:92]:
```suggestion
  # basePlugins are plugins installed and required by the operator
```

File: chart/jenkins-operator/values.yaml [hunk 54]
```diff
@@ -1,70 +1,255 @@
...
+  #   version: "1.76"
+  # - name: configuration-as-code
+  #   version: "1.29"
+  # - name: configuration-as-code-support
+  #   version: "1.19"
+  # - name: kubernetes-credentials-provider
+  #   version: 0.12.1
+  basePlugins: []
+
+  # plugins are plugins installed by user
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:118]:
```suggestion
  # plugins are plugins required by the user
```

File: chart/jenkins-operator/values.yaml [hunk 55]
```diff
@@ -1,70 +1,255 @@
...
+
+      # size is size of PVC
+      size: 5Gi
+
+      # className is storageClassName for PVC
+      # See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1 for more details
+      className: ""
+
+    # env contains container environment variables
+    # Operator handles these variables:
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:202]:
```suggestion
    # PVC backup provider handles these variables:
```

File: website/content/en/docs/Installation/_index.md [hunk 56]
```diff
@@ -43,3 +47,14 @@ kubectl get pods -w
 ```
 
 Now **Jenkins Operator** should be up and running in the `default` namespace.
+
+### Using Helm Chart
+
+There is a option to use Helm to install the operator. It requires the Helm 3+ for deployment.
```
Comment (antoniaklja) [inline website/content/en/docs/Installation/_index.md:53]: There is **an** option to use **the** Helm to install the operator.

File: website/content/en/docs/Installation/_index.md [hunk 57]
```diff
@@ -43,3 +47,14 @@ kubectl get pods -w
...
+
+To install, you need only to type these commands:
```
Comment (antoniaklja) [inline website/content/en/docs/Installation/_index.md:55]: To install, you need only to type these commands: -> The chart can then be installed directly:

File: website/content/en/docs/Installation/_index.md [hunk 58]
```diff
@@ -30,6 +30,10 @@ kubectl apply -f https://raw.githubusercontent.com/jenkinsci/kubernetes-operator
 
 ## Deploy Jenkins Operator
 
+There are two ways to deploy the Jenkins Operator.
+
+### Using YAML's
```
Comment (antoniaklja) [inline website/content/en/docs/Installation/_index.md:35]: Using YAML's -> Using Kubernetes Manifests

**Pull request merged**